### PR TITLE
dont use OnCommitCallback

### PR DESCRIPTION
**Pull request by agavra**
the on commit callback is only called when we flush anyway, so this ultimately caused us to call the `CommitBuffer#flush` method twice back to back leading to confusing statements cluttering the log (empty buffers)

### CODE REVIEW

**Review by apurvam — APPROVED** [review]
I think you should probably wait for another review if you want validation of the reasoning.

**Review by rodesai — APPROVED** [review]
LGTM

**Pull request merged**
